File: 1-10/project10.py
#ProjectEuler Number 9
#by Josh Smith

import math

def is_prime(n):
    if n % 2 == 0 and n > 2: 
        return False
    return all(n % i for i in range(3, int(math.sqrt(n)) + 1, 2))
    
total = 0
for i in range(2, 2000000):
	if(is_prime(i) == True):
		total += i
		
print total

File: C++/project10.cpp
//ProjectEuler Number 10
//by Josh Smith

#include <iostream>
#include <cmath>
using namespace std;

bool isPrime(size_t n){
	if(n < 0){
		cout << "n must be larger than zero" << endl;
		return false;
	}

	for(int i = 2; i < sqrt(n)+1; i++){
		if(n%i == 0){
			return false;
		}
	}
	return true;
}

int main() {
	
	size_t sum = 2;
	size_t x = 3;
	while(x <= 2000000){
		if(isPrime(x)){
			sum += x;
		}
		x++;
	}
	cout << sum << endl;
	return 0;
}
